catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileAlterationObserver.java

```diff
@@ -383,7 +387,12 @@ private void checkAndNotify(
 
     int c = 0;
     for (final AsyncFileEntry entry : previous) {
-      if (c < files.length && !expiredNotUpdated(files[c])) {
+
+      while (c < files.length && failedAndNotUpdated(files[c])) {
+        c++;
+      }
+
+      if (c < files.length && !failedAndNotUpdated(files[c])) {
```

Comment:
Do we need that second condition, `!failedAndNotUpdated(files[c])`? It seems like the while loop before this would only satisfy the first condition (c < file length) if line 391's failedAndNotUpdated were false

Reply:
Nope. We can remove the whole if statement now.